src/renderer/lib/dolphin/useDolphinListeners.ts

```diff
@@ -35,8 +41,9 @@ export const useDolphinListeners = (dolphinService: DolphinService) => {
     }
   }, []);
 
-  const dolphinCompleteHandler = useCallback((event: DolphinDownloadCompleteEvent) => {
+  const dolphinCompleteHandler = useCallback(async (event: DolphinDownloadCompleteEvent) => {
```

Comment:
```suggestion
  const dolphinCompleteHandler = useCallback((event: DolphinDownloadCompleteEvent) => {
```
this probably doesn't still need to be asynchronous